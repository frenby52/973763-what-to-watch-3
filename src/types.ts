export type Film = {
  title: string;
  posterImage: string;
  previewImage: string;
  backgroundImage: string;
  backgroundColor: string;
  description: string;
  ratingScore: number;
  ratingCount: number;
  director: string;
  starring: string[];
  runTime: string;
  genre: string;
  releaseDate: number;
  id: number | string;
  isFavorite: boolean;
  videoLink: string;
  previewSrc: string;
};

export type Comment = {
  id: string;
  user: {
    id: number;
    name: string;
  }
  rating: number;
  text: string;
  date: object;
};
