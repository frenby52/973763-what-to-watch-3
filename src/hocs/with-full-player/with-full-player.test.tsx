import * as React from "react";
import * as renderer from "react-test-renderer";
import withFullPlayer from "./with-full-player";
import {Film} from "../../types";

const promoFilm: Film = {
  id: 0,
  title: `The Grand Budapest Hotel`,
  previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  genre: `Drama`,
  releaseDate: 2014,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  ratingScore: 8.9,
  ratingCount: 240,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  runTime: `20`,
  backgroundColor: `black`,
  isFavorite: false,
  videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};

interface MockComponentProps {
  children: React.ReactNode | React.ReactNode[];
}

const MockComponent = (props: MockComponentProps) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const MockComponentWrapped = withFullPlayer(MockComponent);

it(`withFullPlayer HOC is rendered correctly`, () => {
  const tree = renderer.create(<MockComponentWrapped autoPlay={false} film={promoFilm} muted={false}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
