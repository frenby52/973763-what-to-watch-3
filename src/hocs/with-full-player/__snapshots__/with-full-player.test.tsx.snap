// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withFullPlayer HOC is rendered correctly 1`] = `
<div>
  <video
    autoPlay={false}
    className="player__video"
    muted={false}
    onClick={[Function]}
    onLoadedMetadata={[Function]}
    onTimeUpdate={[Function]}
    poster="the-grand-budapest-hotel-poster.jpg"
    src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  />
</div>
`;
